### ISSUE TITLE
Can't create Popover at primaryAction

### ISSUE THREAD

**Issue by monodisco** [Shopify/polaris-react#2679]
## Issue summary

I'm trying to create a popover in my primaryAction button. 

## Expected behavior

I want to show a popover when I click on the primaryAction button.

## Actual behavior

This is a fragment of my code

`      const primaryAction = {
        content: "Settings",
        onAction: this.togglePopover
      };

    return (
      <Page
        title="Test"
        primaryAction={primaryAction}
      >
        <Popover
          fullHeight
          active={this.state.datePickerActive}
          activator={primaryAction}
          onClose={this.togglePopover}
          sectioned
        >
          <ActionList
            items={[
              { content: "Today", onAction: this.handleToday },
              { content: "Yesterday", onAction: this.handleYesterday },
              { content: "This week", onAction: this.handleWeek },
              { content: "This month", onAction: this.handleMonth }
            ]}
          />
        </Popover>`

This is giving me a "Error: React.Children.only expected to receive a single React element child."

But if I create a button like:

`const activator = <Button onClick={this.togglePopover}>Date picker</Button>;`

...and set `activator={activator}` in the Popover, it works as I expect.

How can I fix this?

**Comment by LauraAubin** [comment]
👋Hello!

The issue here is that you're trying to set the activator to an object, not an actual element (the activator type is `React.ReactElement`).

Since the object `primaryAction` has two parameters, I believe this is why you're receiving an error related to single children because JSX can only render one element.

It's currently not possible to accomplish what you're trying to do because of the way we render the button from `Page`, since there is no way to capture a reference to the `primaryAction` button (which you could hypothetically pass as the activator).

We recommend against doing this, see the [Best Practices section](https://polaris.shopify.com/components/overlays/popover) for more info,

> Always be positioned next to the button or other interface element that triggers them

<br>

I hope this helps answer your question. In your particular use case, perhaps you could avoid using the page to launch a popover and instead, create a specific activator button?

**Comment by monodisco** [comment]
Hi!

Thanks for your response. If it's not possible to capture the reference from the `primaryAction`, I will do it by creating a specific activator button as you sugested:

```
const activator = <Button onClick={this.togglePopover}>Date picker</Button>;

  return(
  <Page
    title="Test"
  >
    <Popover
      fullHeight
      active={this.state.datePickerActive}
      activator={activator}
      onClose={this.togglePopover}
      sectioned
    >
      <ActionList
        items={[
          { content: "Today", onAction: this.handleToday },
          { content: "Yesterday", onAction: this.handleYesterday },
          { content: "This week", onAction: this.handleWeek },
          { content: "This month", onAction: this.handleMonth }
        ]}
      />
    </Popover>
```

**Comment by chloerice** [comment]
Hey @monodisco 👋 It's unclear to me what the actions are that merchants would be taking in your primary action/action list. From the code snippet it sounds like more of a select or a choice list form input. 

That said, a split variant of `Button` shipped with v4.12.0. So the object you set on `Page.primaryAction` now supports an optional `connectedDisclosure` property  ([see the "Split button" example in the style guide](https://polaris.shopify.com/components/actions/button)). Use this when there is only one primary action but other related actions can be taken.

**Comment by monodisco** [comment]
Thanks for your approach @chloerice . What I want to do is a Popover with different time selectors (and also a DatePicker) that updates a table.

I couldn't make it work with the primary action, but with a separate button works as expected.
